### PR TITLE
SQL-1275: add M1 build tasks

### PR DESCRIPTION
**Pull request by terakilobyte**
I took this over from Patrick while he's out on vacation, so 99% of the credit goes to him!

### CODE REVIEW

File: resources/run_adf.sh [hunk 1]
```diff
@@ -210,7 +221,7 @@ install_mongosh() {
       chmod -R +x $LOCAL_INSTALL_DIR/$MONGOSH_UNZIP_DIR/bin/
       echo $LOCAL_INSTALL_DIR/$MONGOSH_UNZIP_DIR
     else
-      tar zxf $LOCAL_INSTALL_DIR/$MONGOSH_DOWNLOAD_FILE --directory $LOCAL_INSTALL_DIR
+      unzip -qo $LOCAL_INSTALL_DIR/$MONGOSH_DOWNLOAD_FILE -d $LOCAL_INSTALL_DIR 2> /dev/null
```
Comment (terakilobyte) [inline resources/run_adf.sh:224]: While `tar` works locally, it wasn't working on the build host in evergreen! This file downloads as a `.zip`, so I used `unzip`.

**Review by mattChiaravalloti — APPROVED** [review]
Looks good! 💯

File: resources/run_adf.sh (hunk 1 above)
Comment (pmeredit) [reply resources/run_adf.sh:225]: That would explain the issue I was seeing!

**Review by pmeredit — APPROVED** [review]
Looks great! Thanks for getting it across the line!

**Comment by terakilobyte** [comment]
evergreen merge

**Pull request merged**